src/impl.jl

```diff
@@ -249,16 +264,34 @@ function auto_hash_equals_impl(__source__, struct_decl, fields, cache::Bool, has
             x._cached_hash
         end)))
     else
-        hash_typearg = typearg && !isnothing(where_list)
-        hash_init = hash_typearg ? :($hashfn($full_type_name, h)) : :($hashfn($(QuoteNode(type_name)), h))
+        hash_init =
+            if isnothing(typeseed)
+                if typearg
+                    :($type_seed($full_type_name, h))
```

Comment:
Could we mix in `h` via addition and pull out the `type_seed` computation to macro runtime?

Reply:
No, because we don't know the runtime type until runtime.

Reply:
Oh right, in some performance sensitive places we `eval` to get the runtime argument types, but here we're literally in the process of defining the type 😅

Reply:
We also don't know the (runtime) values of the type parameters.